compendium/CM/include/cppmicroservices/cm/ConfigurationListener.hpp

```diff
@@ -26,109 +26,124 @@
 #include "cppmicroservices/ServiceReference.h"
 #include "cppmicroservices/cm/ConfigurationAdmin.hpp"
 
-namespace cppmicroservices {
-namespace service {
-namespace cm {
-/**
- \defgroup gr_configurationlistener ConfigurationListener
- \brief Groups ConfigurationListener class related symbols.
- */
-/**
- * \addtogroup gr_configurationlistener
- * @{
- * The ConfigurationEventType is passed to a Configuration Listener to 
- * identify the type of ConfigurationEvent that has occurred.
- */
-enum class ConfigurationEventType
+namespace cppmicroservices
 {
-  CM_UPDATED =
-    1, ///< The ConfigurationEvent type for when a Configuration object has been updated
-  CM_DELETED =
-    2 ///< The ConfigurationEvent type for when a Configuration object has been removed
-};
-/** @}*/
+    namespace service
+    {
+        namespace cm
+        {
+            /**
+             \defgroup gr_configurationlistener ConfigurationListener
+             \brief Groups ConfigurationListener class related symbols.
+             */
+            /**
+             * \addtogroup gr_configurationlistener
+             * @{
+             * The ConfigurationEventType is passed to a Configuration Listener to
+             * identify the type of ConfigurationEvent that has occurred.
+             */
+            enum class ConfigurationEventType
+            {
+                CM_UPDATED = 1, ///< The ConfigurationEvent type for when a Configuration object has been updated
+                CM_DELETED = 2  ///< The ConfigurationEvent type for when a Configuration object has been removed
+            };
+            /** @}*/
 
-/**
- * \ingroup gr_configurationlistener
- *
- * The ConfigurationEvent object is passed to the ConfigurationListener when 
- * the configuration object for any service is updated or removed by ConfigurationAdmin
- */
-class ConfigurationEvent
-{
-public:
-  ConfigurationEvent(const ServiceReference<ConfigurationAdmin> configAdmin,
-                     const ConfigurationEventType type,
-                     const std::string factoryPid,
-                     const std::string pid)
-    : configAdmin(std::move(configAdmin))
-    , type(type)
-    , factoryPid(std::move(factoryPid))
-    , pid(std::move(pid))
-  {}
+            /**
+             * \ingroup gr_configurationlistener
+             *
+             * The ConfigurationEvent object is passed to the ConfigurationListener when
+             * the configuration object for any service is updated or removed by ConfigurationAdmin
+             */
+            class ConfigurationEvent
+            {
+              public:
+                ConfigurationEvent(ServiceReference<ConfigurationAdmin> const configAdmin,
```

Comment:
We should fix these before submission. It doesn't make any sense to do a std::move on a const object.

Reply:
@carneyweb please create a github issue for this